src/gen9_avc_encoder.c

```diff
@@ -5634,17 +5910,1000 @@ gen9_avc_kernel_sfd(VADriverContextP ctx,
+        cmd->dw54.roi_3_x_right  = generic_state->roi[2].right;
+        cmd->dw54.roi_3_y_bottom = generic_state->roi[2].bottom;
+
+        cmd->dw55.roi_4_x_left   = generic_state->roi[3].left;
+        cmd->dw55.roi_4_y_top    = generic_state->roi[3].top;
+        cmd->dw56.roi_4_x_right  = generic_state->roi[3].right;
+        cmd->dw56.roi_4_y_bottom = generic_state->roi[3].bottom;
+
+        cmd->dw36.enable_cabac_work_around = 0;
+
+        if (!generic_state->brc_enabled) {
+            char tmp = 0;
+            tmp = generic_state->roi[0].value;
+            CLIP(tmp, -qp, AVC_QP_MAX - qp);
+            cmd->dw57.roi_1_dqp_prime_y = tmp;
+            tmp = generic_state->roi[1].value;
+            CLIP(tmp, -qp, AVC_QP_MAX - qp);
+            cmd->dw57.roi_2_dqp_prime_y = tmp;
+            tmp = generic_state->roi[2].value;
+            CLIP(tmp, -qp, AVC_QP_MAX - qp);
+            cmd->dw57.roi_3_dqp_prime_y = tmp;
+            tmp = generic_state->roi[3].value;
+            CLIP(tmp, -qp, AVC_QP_MAX - qp);
+            cmd->dw57.roi_4_dqp_prime_y = tmp;
+        } else {
+            cmd->dw34.roi_enable_flag = 0;
+        }
+    }
+
+    cmd->dw65.mb_data_surf_index = GEN8_AVC_MBENC_MFC_AVC_PAK_OBJ_CM;
+    cmd->dw66.mv_data_surf_index =  GEN8_AVC_MBENC_IND_MV_DATA_CM;
+    cmd->dw67.i_dist_surf_index = GEN8_AVC_MBENC_BRC_DISTORTION_CM;
+    cmd->dw68.src_y_surf_index = GEN8_AVC_MBENC_CURR_Y_CM;
+    cmd->dw69.mb_specific_data_surf_index = GEN8_AVC_MBENC_MB_SPECIFIC_DATA_CM;
+    cmd->dw70.aux_vme_out_surf_index = GEN8_AVC_MBENC_AUX_VME_OUT_CM;
+    cmd->dw71.curr_ref_pic_sel_surf_index = GEN8_AVC_MBENC_REFPICSELECT_L0_CM;
+    cmd->dw72.hme_mv_pred_fwd_bwd_surf_index = GEN8_AVC_MBENC_MV_DATA_FROM_ME_CM;
+    cmd->dw73.hme_dist_surf_index = GEN8_AVC_MBENC_4xME_DISTORTION_CM;
+    cmd->dw74.slice_map_surf_index = GEN8_AVC_MBENC_SLICEMAP_DATA_CM;
+    cmd->dw75.fwd_frm_mb_data_surf_index = GEN8_AVC_MBENC_FWD_MB_DATA_CM;
+    cmd->dw76.fwd_frm_mv_surf_index = GEN8_AVC_MBENC_FWD_MV_DATA_CM;
+    cmd->dw77.mb_qp_buffer = GEN8_AVC_MBENC_MBQP_CM;
+    cmd->dw78.mb_brc_lut = GEN8_AVC_MBENC_MBBRC_CONST_DATA_CM;
+    cmd->dw79.vme_inter_prediction_surf_index = GEN8_AVC_MBENC_VME_INTER_PRED_CURR_PIC_IDX_0_CM;
+    cmd->dw80.vme_inter_prediction_mr_surf_index = GEN8_AVC_MBENC_VME_INTER_PRED_CURR_PIC_IDX_1_CM;
+    cmd->dw81.flatness_chk_surf_index = GEN8_AVC_MBENC_FLATNESS_CHECK_CM;
+    cmd->dw82.mad_surf_index = GEN8_AVC_MBENC_MAD_DATA_CM;
+    cmd->dw83.force_non_skip_mb_map_surface = GEN8_AVC_MBENC_FORCE_NONSKIP_MB_MAP_CM;
+    cmd->dw84.widi_wa_surf_index = GEN8_AVC_MBENC_WIDI_WA_DATA_CM;
+    cmd->dw85.brc_curbe_surf_index = GEN8_AVC_MBENC_BRC_CURBE_DATA_CM;
+    cmd->dw86.static_detection_cost_table_index = GEN8_AVC_MBENC_STATIC_FRAME_DETECTION_OUTPUT_CM;
+
+    i965_gpe_context_unmap_curbe(gpe_context);
+
+    return;
+}
+
+static void
+gen8_avc_send_surface_mbenc(VADriverContextP ctx,
```

Comment:
this function is almost same as gen9_avc_send_surface_mbenc. is it possible to reuse gen9_avc_send_surface_mbenc?